### DISCUSSION TITLE
FriendlyElec - Ubuntu 22.04 - Kernel 6.1 - Wayland - w/ Mali blob drivers

### DISCUSSION THREAD

**Discussion by hubby4** [Joshua-Riek/ubuntu-rockchip#447] [Ideas]
In issue post #443, github member Nyanmisaka linked to a wiki page describing a custom Ubuntu 22.04 build by FriendlyElec for their RK3588 boards that supports Wayland-based Mali blob drivers instead of panfork. Nyanmisaka also claims that this build doesn't have the "green video" issue that plagues the panfork mesa drivers. I tried to install this build on my Orange Pi 5, but with no luck.

Here's a link to the wiki page explaining the features of the build.
https://wiki.friendlyelec.com/wiki/index.php/NanoPC-T6#Work_with_Ubuntu_22.04_Desktop

Here's a link to the google drive page, which hosts the official FriendlyElec RK3588 image files, the source files, and documents.
https://drive.google.com/drive/folders/1FoBbP_nPkMehwBj4wHwsbRU-QGjEdeEP

I'd like to be able to run this build on my Orange Pi 5 if it's possible. Any assistance is appreciated.

**Comment by Joshua-Riek** [comment]
This kernel is in development and not yet released by rockchip. I have direct access to the internal sdk servers and have been watching this closely. 

FriendlyElecs choice to bump their kernel up to a work in progress dev branch is immature. I've already tested this kernel, and it has quirks. Additionally, it will take a lot of work to rebase all patches from 5.10 onto 6.1 it's not at all a simple process.

**Reply by hubby4** [reply]
Good to know. In that case, I'll stick with this current release. Aside from the green video issues, 5.10 + Panfork + custom patches is fairly solid.

**Reply by nyanmisaka** [reply]
They used to have wayland images using 5.10 kernel, but the archives were removed due to the addition of the new version. Maybe there's no google drive spaces?

I learned that the image were built using buildroot/ycoto. This means that its built-in wayland compositor may contain patches from RK to make libmali+wayland work properly on Gnome.
